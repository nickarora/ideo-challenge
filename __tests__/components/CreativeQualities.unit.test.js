jest.mock('assets/images/qualityIcons/collaboration.png', () => 'img')
jest.mock('assets/images/qualityIcons/empowerment.png', () => 'img')
jest.mock('assets/images/qualityIcons/purpose.png', () => 'img')

const mockFetchResult = mockResponse => {
  jest.doMock('react-refetch', () => {
    class PromiseState {
      constructor(promiseStateConfig) {
        this.pending = promiseStateConfig.pending
        this.rejected = promiseStateConfig.rejected
        this.reason = promiseStateConfig.reason
        this.value = promiseStateConfig.value
      }
    }

    return {
      connect: () => Component => () => (
        <Component creativeQualitiesFetch={new PromiseState(mockResponse)} />
      ),
      PromiseState,
    }
  })

  return require('~/components/CreativeQualities').default // eslint-disable-line
}

beforeEach(() => {
  jest.resetModules()
})

describe('CreativeQualities', () => {
  it('renders a loading animation if the fetch request is still pending', () => {
    const CreativeQualities = mockFetchResult({
      pending: true,
      rejected: false,
      reason: null,
      value: null,
    })

    const wrapper = shallow(<CreativeQualities />).dive()
    expect(wrapper).toMatchSnapshot()
  })

  it('renders an error if the fetch request fails', () => {
    const CreativeQualities = mockFetchResult({
      pending: false,
      rejected: true,
      reason: 'unauthorized',
      value: null,
    })

    const wrapper = shallow(<CreativeQualities />).dive()
    expect(wrapper).toMatchSnapshot()
  })

  it('renders creative qualities once data has arrived', () => {
    const CreativeQualities = mockFetchResult({
      pending: false,
      rejected: false,
      reason: null,
      value: [],
    })

    const wrapper = shallow(<CreativeQualities />).dive()
    expect(wrapper).toMatchSnapshot()
  })
})
