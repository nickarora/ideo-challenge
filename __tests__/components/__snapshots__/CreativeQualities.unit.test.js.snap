// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreativeQualities renders a loading animation if the fetch request is still pending 1`] = `<LoadingAnimation />`;

exports[`CreativeQualities renders an error if the fetch request fails 1`] = `
<Error
  error="unauthorized"
/>
`;

exports[`CreativeQualities renders creative qualities once data has arrived 1`] = `
<Connect(Qualities)
  qualities={Array []}
/>
`;
